chapters/common_pitfalls.md

```diff
@@ -0,0 +1,44 @@
+# Common Pitfalls for new Vulkan Developers
+
+This is a short list of common assumptions and traps developers new to Vulkan can make.  
+
+### Validation Layers
+
+During development, ensure that the Validation Layers are enabled. They are an invaluable tool for catching mistakes while using the Vulkan API. Parameter checking, object lifetimes, and threading violations all are part of the provided error checks. A way to reassure that they are enabled is to verify if the text "Debug Messenger Added" is in the output stream. More info can be found in the [Vulkan SDK](https://vulkan.lunarg.com/doc/sdk/latest/windows/layer_configuration.html) layer documentation. It is first on the list because it is the first line of defense and should help reduce the time spent discovering where the mistake was made.
+
+### Vulkan is a box of tools
+
+In Vulkan, there are often multiple solutions to a problem. Consider passing data to a shader, one can use uniform buffers, storage buffers, vertex attributes, instance buffer, push constants, and more. Are all valid methods each with their own benefits and drawbacks. There is rarely a "perfect" solution to any problem. Therefore, use the tools available and create a good solution instead of trying to find a perfect solution.
```

Comment:
This paragraph is a little weird without at least some pointer as to how to judge the best tool for the job. I think some indication of where to find resources to help (e.g. this guide!) is a good idea.

Reply:
Reworked this but not sure if it addressed the commented issues.